File: consultas/consultas/report/detalle_de_ventas_betalab/detalle_de_ventas_betalab.js
// Copyright (c) 2016, Lewin Villar and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Detalle de Ventas Betalab"] = {
	"filters": [
		{
			"label": __("From Date"),
			"fieldname": "from_date",
			"fieldtype": "Date",
			"default": frappe.datetime.year_start(),
			"reqd": 1,
		},
		{
			"label": __("To Date"),
			"fieldname": "to_date",
			"fieldtype": "Date",
			"default": frappe.datetime.year_end(),
			"reqd": 1,
		},
		// {
		// 	"label": __("Sucursal"),
		// 	"fieldname": "sucursal",
		// 	"fieldtype": "Link",
		// 	"options": "Empresa",
		// 	"reqd": 1,
		// },
		{
			"label": __("Prueba"),
			"fieldname": "prueba",
			"fieldtype": "Link",
			"options": "Prueba",
			"reqd": 1,
		},
		{
			"label": __("Tipo de Consulta"),
			"fieldname": "tipo_de_consulta",
			"fieldtype": "Select",
			"options": "\nConsulta Privada\nConsulta Seguro",
		},
		{
			"label": __("Team"),
			"fieldname": "team",
			"fieldtype": "Link",
			"options": "Institucion",
		},
	],
	formatter: function (row, cell, value, columnDef, dataContext, default_formatter) {
		value = default_formatter(row, cell, value, columnDef, dataContext);
		
		if (cell == 1) {
			let dt = dataContext['Documento'].split("-")[0] == "CLP" ? "Consulta Privada" : "Consulta Seguro" ;
			value = `<a class="grey" target="_blank" href="#Form/${dt}/${dataContext['Documento']}">${dataContext['Documento']}</a>`;
		}

		return value;
	}
}

File: consultas/consultas/report/detalle_de_ventas_betalab/detalle_de_ventas_betalab.py
# Copyright (c) 2013, Lewin Villar and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe.utils import flt

def execute(filters=None):
	return get_columns(), get_data(filters)

def get_columns():
	columns = [
		"Documento:Data:120",
		"Tipo:Data:120",
		"Fecha:Date:100",
		"Sucursal:Data:220",
		"Paciente:Data:200",
		"Precio:Currency:120",
	]

	return columns

def get_conditions(filters):

	conditions = ["v.docstatus in (0,1)"]

	
	if filters.get("tipo_de_consulta"):
		conditions.append("v.tipo_de_consulta = '{}'".format(filters.get("tipo_de_consulta")))
	# if filters.get("sucursal"):
	conditions.append("v.sucursal  in ('EMP-000006', 'EMP-000014', 'EMP-000012', 'EMP-000018', 'EMP-000003')")
	
	if filters.get("prueba"):
		conditions.append("v.prueba = '{}'".format(filters.get("prueba")))	
	if filters.get("from_date"):
		conditions.append("v.fecha >= '{}'".format(filters.get("from_date")))
	if filters.get("to_date"):
		conditions.append("v.fecha <= '{}'".format(filters.get("to_date")))
	if filters.get("team"):
		conditions.append("v.team = '{}'".format(filters.get("team")))
	
	return " AND ".join(conditions)

def get_data(filters):
	conditions = get_conditions(filters)
	results = []
	return frappe.db.sql("""
		SELECT 
			v.parent, 
			v.tipo_de_consulta,
			v.fecha,
			v.sucursal_nombre,
			v.paciente,
			v.monto
		FROM 
			`viewVentas Por Prueba` v
		WHERE 
			{conditions}
		GROUP BY 
			v.parent
		ORDER BY 
			v.fecha
		""".format(conditions=conditions), debug=True)

	